uefi/src/table/boot.rs

```diff
@@ -2082,6 +2079,104 @@ pub struct MemoryMapSize {
     pub map_size: usize,
 }
 
+/// An iterator of [`MemoryDescriptor`] which returns elements in sorted order. The underlying memory map is always
+/// associated with the unique [`MemoryMapKey`] contained in the struct.
+pub struct MemoryMap<'buf> {
+    key: MemoryMapKey,
+    buf: &'buf mut [u8],
+    entry_size: usize,
+    len: usize,
+}
+
+impl<'buf> MemoryMap<'buf> {
+    #[allow(clippy::must_use_candidate)]
```

Comment:
Rather than adding an allow for `must_use_candidate` here and elsewhere, add `#[must_use]`.